### PR TITLE
EnvoyProxy Auth Layer POC

### LINKED ISSUE #24: Auth-Layer: EnvoyProxy POC
```[tasklist]
### Tasks
- [x] Validate is a valid Json
- [x] Extract Token, method and params
- [x] Check Token on Postgres DB
- [x] Check Token on Redis
- [x] Push PR for Reviews
```

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
This is a good start for the implementation of `EnvoyProxy` as a reverse proxy for `Hedera-The-Graph` Auth layer.

**Related issue(s)**:
#18 

Fixes #24

**Notes for reviewer**:

Next tasks:
- Decide if we move forward using `EnvoyProxy`.
     (I would say Yes)
- Decide if we will use `redis` vs `postgres` for the token db store.
    (I would say postgres)
- Clean the unneeded code for the option we discarded.
- Decide if we want to Hash the token on the database, and add the function to the lua script
- Normalize the database, do a final implementation/design of the database and its scripts
- Create a new Chart for `hedera-the-graph-auth-layer`
- Integrate new Chart on the umbrella Chart for the overall deployment of `hedera-the-graph`.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: envoy-auth-layer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM    envoyproxy/envoy:v1.28-latest
+# copy envoy.yaml to the container
```
Comment (Nana-EC) [inline envoy-auth-layer/Dockerfile:2]: nit: remove commented out code

File: envoy-auth-layer/README.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+it includes a Dockerfile for building the image and a docker-compose file for running the container.
+
+## Pre-requisites
+
+### Postgres
+```
+docker run --name postgres-envoy-test -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres
+```
+
+Run init script to create the database and the table
```
Comment (Nana-EC) [inline envoy-auth-layer/README.md:43]: Consideration for later, we can make the envoy proxy have read only permissions to the db or cache, while the token manager preserves the write permissions

File: envoy-auth-layer/README.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+CREATE SCHEMA auth
+    AUTHORIZATION postgres;
+
+CREATE TABLE IF NOT EXISTS auth.permissions
+(
+    id integer,
+    token character varying(255) NOT NULL,
+    method character varying(50) NOT NULL,
+    param_name character varying(50) NOT NULL
```
Comment (Nana-EC) [inline envoy-auth-layer/README.md:61]: `param_name` is a placeholder for subgraph name and email for later right?

**Review by Nana-EC — COMMENTED** [review]
Nice work.
Jotted down some consideration for post POC

File: envoy-auth-layer/README.md (hunk 3 above)
Comment (AlfredoG87) [reply envoy-auth-layer/README.md:61]: is more like param.name of the json payload. but this is not final, so this could change.

**Pull request closed without merging**